Compact sidebar toggle

RaitaroH opened the pull request.

RaitaroH committed:
compact sidebar toggle

RaitaroH commented:
Forgot to add `https://discordapp.com` for the assets svg. It was working fine but seems that in ice the icon is missing.

the-j0k3r commented:
@RaitaroH One version bump is enough :)

So the servers/channels are not being compacted only the DM windows, right?

RaitaroH commented:
@the-j0k3r well, not really. The selector that decides the width is the same for both channels and servers so... I will try to have a compact hover effect maybe.

RaitaroH commented:
@the-j0k3r :sweat_smile: this PR ended up a bit bigger than I imagined at first. Sorry for the bump version but I always increase the number with every change.

the-j0k3r commented:
>this PR ended up a bit bigger than I imagined at first.

Thats fine :)

>Sorry for the bump version but I always increase the number with every change.

No problem, But if I were to merge this as is (and not bump it myself after) neither version bump on this PR would be in a position that would include all the commits in that version. In order for it to work, the version bump needs to be the last commit.

My only suggestion is that you do an interactive rebase and remove all version bumps except for one and reposition it as the last commit, if you add more commits then rinse and repeat.

RaitaroH committed:
improved compact mode + servers fix

RaitaroH committed:
hover display effect for compact mode

RaitaroH commented:
@the-j0k3r Everything is nice now :yum: 

I will leave the version bump to you then.

the-j0k3r commented:
Alrighty then :) thanks for that cleanup and reorganization, this needs more testing and giving it a chance to have more eyes on this.

@vednoc Would you mind testing and sharing your feedback?

RaitaroH commented:
@vednoc Would you mind fixing the hover over the container (where you have mute button and settings) :angel:

vednoc commented on discord-inspired-bits-dark.user.css:
```diff
@@ -333,4 +333,120 @@
+			color: #fff !important;
+		}
+		/*resizing the chanels*/
+		[class*="channels"]:hover, .container-2Thooq:hover [class*="channels"] {
```
Needs `[class*="channels"]:hover > [class*="private"]` so DM content expands with the parent element. Perhaps remove `.container-2Thooq:hover [class*="channels"]` unless it's needed for something; it wasn't used at all in my testing.

vednoc commented on discord-inspired-bits-dark.user.css:
```diff
@@ -333,4 +333,120 @@
+		}
+		/*hide scrollbar*/
+		[class*="channels"] .scroller-2FKFPG {
+			overflow: hidden !important;
```
This line prevents vertical scrolling and disables the ability to scroll through channels, both in guilds and in direct messages. I noticed the horizontal scrollbar showing up when hover is enabled so `overflow-x` fixes it.

RaitaroH replied:
Forgot the `x`. I barely have any contacts so I missed that.

vednoc reviewed (changes requested):
My notifications are so messed up. :slightly_frowning_face:

I like it! The only thing I would add is `transition-delay` when on hover is enabled. It's quite useful if you want to click through channels and not have them expand immediately; it would also be great if there was a custom setting so we can specify the delay more specifically.

RaitaroH committed:
enable scrolling down, but not sideways

RaitaroH committed:
container hover fix

RaitaroH committed:
hover delay

RaitaroH commented:
@vednoc I've added a transition delay to everything... besides the search icon and `#` in servers. Not sure what I'm missing but it changes immediately on channels hover.

Also not sure if .5s is enough time delay. I thinks it's ok but I could increase it if you want.

I'm terrible at transitions :disappointed:

vednoc commented:
@RaitaroH I'll take https://github.com/style-it-themes/discord-inspired-bits-dark/pull/8/commits/2c7e22190f3812a58324ae46e4c07052c4142baf as a base and add transitions since the https://github.com/style-it-themes/discord-inspired-bits-dark/pull/8/commits/f7a569b0d179291a36e10874a4527c988912d281 is quite buggy in direct messages. 500ms should be alright but I'll add a custom option for it. :smile:

RaitaroH commented:
I suppose I can just rewrite the transitions from scratch because I kinda got lost a bit.

I can add an option for the transition delay too.

vednoc commented:
It actually works pretty alright, turns out I had something changed in the DevTools and forgot to reload the page after installing the update. :roll_eyes: 

An option for transition-delay would be great, but rewriting it from the scratch will make it more or less the same. I have an idea how to do it in a slightly different way but it'll have to wait until later tonight or tomorrow.

RaitaroH committed:
less buggy transitions

RaitaroH committed:
option for transition delay

RaitaroH commented:
@vednoc I made the code a bit less redundant.
+ I am still not sure how to fix the background-image changing of the search tho. I can manage to keep it from becoming dark again but the icon (back-image) disappears and I can also make the text not show up but then you get a blank space. So for now I just left it at that
+ I also fixed the `#` in servers by... not showing them back up :laughing: 
+ I also added a transition delay option as suggested

Also should the right sidebar (in servers) also be collapsible? Roles will have to go tho. I personally have them collapsed to save some space.

vednoc commented:
Looks pretty good! I'll try to fix the background-image of the search bar once this gets merged. Hiding `#` is pretty cool. :smile: 

The right sidebar doesn't have to be collapsible since it can be hidden and you don't interact with it as often, but on small resolutions it could be quite nice to have. Perhaps we could put these settings under `@media` so they get applied without the need to enable/disable them all the time, similar to what I've done in my WhatsApp theme.

RaitaroH commented:
Thanks in advance for the fix.

I had to hide `#` so you could `kinda` read the topics. 

Indeed. So let's leave the right sidebar at that.

RaitaroH commented:
@vednoc @the-j0k3r 

Pretty much this in order to hide the right sidebar too. Should this be in the theme itself or in the compact mode or be toggable in itself? I can add a commit either way.
```css
	/*Compact server sidebar*/
	@media screen and (min-width: 0px) and (max-width: 900px) {
		.membersWrap-2h-GB4 { 
			min-width: 55px !important;
			width: 55px !important;
		}
		/*roles*/
		[class*="membersGroup"] {
			display: none !important;
		}
	}
	@media screen and (min-width: 0px) and (max-width: 600px) {
		.membersWrap-2h-GB4 { 
			display: none !important;
		}
	}
```

RaitaroH commented:
@vednoc do you also get `instant invites` in servers? I have it in firefox but not in waterfox for some reason.
```css
.marginReset-3RfdVe { display: none !important; }
```
this selector should be hidden like `#`

vednoc commented:
@RaitaroH I think a separate toggle is going to be better for most users. Media query is definitely a nice touch and it makes the toggle kinda unnecessary. Choices! :smile: 

It seems like `instant invites` stop showing up in all servers once you close it on any server. :thinking: You can hide it with `[class*="channelNotices"] > [class*="invite"]` and it'll never show up again.

RaitaroH committed:
hide instant invites in compact mode

RaitaroH committed:
server sidebar compact: always & @media

RaitaroH committed:
transition for server sidebar

RaitaroH commented:
@vednoc checkout 2747b9f, cc1e811 & 256418a :wink:

Edit: I do have some ghost empty space at the bottom of the sidebar though. Not exactly sure why. If I scroll ever so slightly the space is filled.

vednoc commented:
@RaitaroH It's amazing! :smiley:

Same here. They use JavaScript to show more users depending on that element's height, similar to how they show users in direct messages. I remember a while back trying to make a more compact view for DM's but it didn't work out all that well. We can either leave the space empty or let the users have slightly more padding/margin applied in order to cover up for the empty space. Your thoughts?

RaitaroH commented:
@vednoc well, I would like not to have extra work to do so... if you just give me the code needed for the cover up that would be just perfect :angel:

vednoc commented:
Absolutely! Had some family time so it got delayed, but the following should do the trick. I've tried a few different methods but this one seems to be the best. :smiley: It should go under both `server_sidebar` sections.

```css
[class*="members"] > [class*="member"] {
  margin: 5px 0 5px 8px;
}
```

Also there are two typos in UserCSS `hvoer` & `chnage` that should be fixed. :smile:

RaitaroH committed:
typo fix

RaitaroH committed:
removing empty space in server sidebar

RaitaroH commented:
@vednoc done and done :yum: 

Seems right now the only thing that is still missing is the proper hover effect for the searchbar. But that's about it I guess.

RaitaroH commented:
@vednoc In discord @the-j0k3r suggested that `Direct Messages` should be invisible because icons would shift a bit. I personally think it should hide itself to save some space. And no... adding one extra option for that is overkill.

What do you think vednoc?

vednoc commented:
>Seems right now the only thing that is still missing is the proper hover effect for the searchbar. But that's about it I guess.

This should fix that, I'll mark added lines with `+`. :smiley: 

```css
[class*="channels"]:hover .private-channels .search-bar .search-bar-inner {
+ background-color: rgba(0,0,0,0.2) !important;
  border: 1px solid rgba(0,0,0,0.1) !important;
  margin-left: 0px !important;
+ padding-left: 20px;
}

.private-channels .search-bar .search-bar-inner {
  background: transparent url("https://discordapp.com/assets/92222a08554d0e19d6bf1a45ca780b7c.svg") no-repeat !important;
  background-position-y: center !important;
+ background-position-x: 5px !important;
  border-color: transparent !important;
  margin-left: 8px !important;
  filter: brightness(200%);
}
```

>@vednoc In discord @the-j0k3r suggested that Direct Messages should be invisible because icons would shift a bit. I personally think it should hide itself to save some space. And no... adding one extra option for that is overkill.

I don't know what to say because I'm fairly confused right now. :thinking: To me, it's good as is.

RaitaroH committed:
search bar hover fix

RaitaroH commented:
@vednoc I added the above in 507fd5e. So I guess now is time for a merge, if you don't have something else to add. 

That space doesn't matter tho. In my personal userContent.css file I just hide that space :yum: 

In other news, I offered some code in issue #11 :yum: , you know, for completeness.

the-j0k3r commented:
Ill get to this, pending some testing, but Ill be really busy this week, so looking at least for next weekend or the following week.

Sorry :(

RaitaroH commented:
@the-j0k3r doesn't bother me man. I am already using the code anyway :yum:

RaitaroH committed:
making sure invites are hidden in servers

the-j0k3r commented:
Hello again @RaitaroH  @vednoc 

Thank you for the hard work on this feature.

I had a look (finally had a minute free) and I have a couple of minor suggestions for option names as follows: (feel free to copy and paste this block bellow.

```
@var checkbox compact_mode 			  "Compact left sidebar" 0 {}
@var select   server_sidebar		  "Compact right sidebar" {
    "On resize": "resize",
    "Always": "compact"
}
@var checkbox compact_mode_transition "Use hover effect for compact mode" 1 {}
@var text     transition_delay 		  "Effect timing on hover" 0.7s
```
**Changes explanation.**

* Renamed options a bit
* Made  also thought that the ""Use hover effect for compact mode" enabled by default, I realize that this option does nothing unless Compact right/left sidebar is on but I think as default this works better. (what do you guys think?).
* Renamed server sidebar to right sidebar. (idk if that was a typo originally?)
* Placed compact options next to each other.

I also like to see the following implemented.

* Channel name preceding `#` symbol to be restored on hover.
* Hide right/right sidebar Group header name so we will have the shifting back and since both of you like it better more compact and my feelings here are not THAT strong. Sorry for being a pain. :)

so @RaitaroH if you rebase this PR on master and be so kind as to review and implement those, I think we can go ahead and merge this asap.

Overall I like this, despite that channel names are cut off which could be worked on in future, but the changes above would make for a better initial implementation for this.

the-j0k3r commented:
And my apologies for delay, but Real Life has been/will remain quite busy for the foreseeable future.

RaitaroH committed:
changed orger of stylus options

RaitaroH commented:
@the-j0k3r Done and done.

RaitaroH commented:
welp... everything is broken now....

vednoc commented:
Finally got around finishing the rewrite of compact mode today with @RaitaroH. Here's a long snippet of code, everything should be working well though there might be that issue with scrollbars in Firefox on Windows.

The snippet replaces everything from line `341` until line `456` and makes a *good* use of stylus-lang syntax. :smiley:

```stylus
	if compact_mode {
		// disable "direct messages" text
		[class*="privateChannels"] header {
			display: none !important;
		}
		[class*="channels"] {
			width: 70px !important;
			transition: 300ms ease-in-out;
			// apply only if CMT is enabled
			if compact_mode_transition {
				// show sidebar
				&:hover {
					width: 240px !important;
					transition-delay: transition_delay;
				}
				// hide usernames until sidebar is activated
				&:hover [class*="channel"] [class*="name"] {
					margin-left: 0;
					transition-delay: 0.3s;
				}
				// show search-bar
				&:hover .search-bar {
					transition-delay: transition-delay !important;
					background-color: var(--color-4a4d53) !important;
				}
				// animation for search-bar
				&:hover .search-bar {
					transition: 0.3s ease-in-out;
					transition-delay: transition_delay !important;
					&:hover {
						transition-delay: 0s !important;
					}
				}
				// hide usernames
				[class*="channel"] [class*="name"] {
					margin-left: 50px;
					transition: margin-left transition_delay ease-in-out;
					transition-delay: transition-delay;
				}
			}
			// remove extra margin on channels
			[class*="channel"] > a {
				margin: 1px 8px 1px 8px !important;
			}
			// remove instant invites
			[class*="channelNotices"] > [class*="invite"] {
				display: none !important;
			}
			// remove # in servers
			[class*="content"] > [class*="marginReset"] {
				// display: none !important;
				max-width: 0 !important;
				opacity: 0 !important;
			}
			// fix icons in direct messages
			[class*="linkButtonIcon"] {
				min-width: 24px;
			}
			// fix scroller in firefox
			[class*="scroller"] {
				padding-right: 0;
			}
			// your avatar and settings
			[class*="container"] > [class*="avatar-small"] {
				margin: 0 16px 0 6px;
			}
			// typing...
			.avatar-small .status[class*="typing"] {
				right: 0px !important;
			}
			// search bar
			[class*="privateChannels"] .search-bar {
				.search-bar-inner {
					background: transparent url(https://discordapp.com/assets/92222a08554d0e19d6bf1a45ca780b7c.svg) no-repeat !important;
					background-position-x: 5px !important;
					background-position-y: center !important;
					border-color: transparent !important;
					margin-left: 8px !important;
					filter: brightness(200%);
					input {
						width: 0px !important;
						min-width: 0px !important;
						opacity: 0 !important;
					}
					if compact_mode_transition {
						&:hover {
							transition: 200ms ease-in-out;
							transition-delay: transition_delay;
						}
						input {
							padding-left: 35px;
							opacity: 1 !important;
						}
					}
				}
			}
		}
	}
```

vednoc commented:
Just found one last derp that needs to be addressed so this snippet does the job, add it after everything from the last snippet. :smile: 

```stylus
	// fix visible text on chromium
	if not compact_mode_transition {
		[class*="privateChannels"] > a > *:first-child {
			margin-right: 22px !important;
		}
	}
```

RaitaroH committed:
rewrite due to discord changes

many thanks to @vednoc

RaitaroH commented:
I fixed the settings (mute and such) in the container appearing on top of the activities and friends divs.

RaitaroH committed:
container settings bad z-index fix

RaitaroH committed:
better settings container fix

RaitaroH commented:
@the-j0k3r one month later...

the-j0k3r commented:
> @the-j0k3r one month later...

Still a hostage of real life :) Ive sent invites out in meanwhile. Apologies for not being more present in meantime.
